test/library/dynamic_resources.go

```diff
@@ -0,0 +1,88 @@
+package library
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/restmapper"
+)
+
+type doFunc func(t *testing.T, unstructured *unstructured.Unstructured, dynamicResourceInterface dynamic.ResourceInterface)
+
+func NoErrorSkipExists(t *testing.T, err error) {
+	if !k8serrors.IsAlreadyExists(err) {
+		require.NoError(t, err)
+	}
+}
+
+func NoErrorSkipNotExisting(t *testing.T, err error) {
+	if !k8serrors.IsNotFound(err) {
+		require.NoError(t, err)
+	}
+}
+
+func do(do doFunc, t *testing.T, assetFunc func(name string) ([]byte, error), filename string, kubeClient kubernetes.Interface, dynamicKubeClient dynamic.Interface) {
+	b, err := assetFunc(filename)
```

Comment:
BTW, the implementation is based on https://gist.github.com/pytimer/0ad436972a073bb37b8b6b8b474520fc